### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by zhorabay**
- [ ] In this branch, I:
- Created class Person with the following:
-> Instance vars: @id, @name, and @age.
-> Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
-> Getters for @id, @name, and @age.
-> Setters for @name and @age.
-> Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
-> Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- Created class Student with the following:
-> Inherits from Person.
-> Constructor extends parent's constructor by adding @classroom and a parameter for it.
-> Method play_hooky that returns "¯\(ツ)/¯".
- Created class Teacher with the following:
-> Inherits from Person.
-> Constructor extends parent's constructor by adding @specialization and a parameter for it.
-> Override can_use_services? so it always returns true.
- Saved each class  in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Person
+  def initialize(name = 'Unknown', age = 5, parent_permission: true)
```
Comment (TedLivist) [inline person.rb]: - [ ] Only `name` and `parent_permission` are optional parameters, `age` is not. Kindly fix that.

  ```
  def initialize(age, name = 'Unknown', parent_permission: true)
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'person'
+
+class Student < Person
+  def initialize(classroom, name = 'Unknown', age = 5, parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (TedLivist) [inline student.rb]: - [ ] The `age` parameter is not optional. Kindly fix that.

- [ ] Also re-arrange the arguments in the `super` method call.

- [ ] The `parent_permission` argument in the `super` method is assigned differently.

  ```
  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'person'
+
+class Teacher < Person
+  def initialize(specialization, name = 'Unknown', age = 5, parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (TedLivist) [inline teacher.rb]: - [ ] The `age` parameter is not optional. Kindly fix that.

- [ ] Also re-arrange the arguments in the `super` method call.

- [ ] The `parent_permission` argument in the `super` method is assigned differently.

  ```
  def initialize(specialization, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @zhorabay,

Good job so far!
✔️ Implementation of classes in different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require 'person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission: parent_permission)
```
Comment (julie-ify) [inline student.rb]: - [ ] Kindly use the correct order of argument as specified by the previous reviewer. Age should come before name to match the order in the person class.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @zhorabay,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @zhorabay  👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
Made sugged changes

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**